src/server.c

```diff
@@ -129,23 +130,61 @@ void serverLogRaw(int level, const char *msg) {
     } else {
         int off;
         struct timeval tv;
-        int role_char;
         pid_t pid = getpid();
 
         gettimeofday(&tv,NULL);
         struct tm tm;
         nolocks_localtime(&tm,tv.tv_sec,server.timezone,server.daylight_active);
-        off = strftime(buf,sizeof(buf),"%d %b %Y %H:%M:%S.",&tm);
-        snprintf(buf+off,sizeof(buf)-off,"%03d",(int)tv.tv_usec/1000);
-        if (server.sentinel_mode) {
-            role_char = 'X'; /* Sentinel. */
-        } else if (pid != server.pid) {
-            role_char = 'C'; /* RDB / AOF writing child. */
-        } else {
-            role_char = (server.masterhost ? 'S':'M'); /* Slave or Master. */
+        switch(server.log_timestamp_format) {
+            case LOG_TIMESTAMP_DEFAULT:
+                const char* timestamp_format = "%d %b %Y %H:%M:%S.";
+                off = strftime(buf,sizeof(buf),timestamp_format,&tm);
+                snprintf(buf+off,sizeof(buf)-off,"%03d",(int)tv.tv_usec/1000);
+                break;
+
+            case LOG_TIMESTAMP_ISO8601:
+                const char* timestamp_format = "%Y-%m-%dT%H:%M:%S.";
+                off = strftime(buf,sizeof(buf),timestamp_format,&tm);
+                snprintf(buf+off,sizeof(buf)-off,"%03d",(int)tv.tv_usec/1000);
+                char tzbuf[6];
+                strftime(tzbuf, sizeof(tzbuf), "%z", &tm);
+                strncat(buf, tzbuf, sizeof(buf) - strlen(buf) - 1);
+                break;
+
+            case LOG_TIMESTAMP_UNIX:
+                snprintf(buf,sizeof(buf),"%ld",tv.tv_sec);
+                break;
+            default:
+                break;
+        }
+
+        switch (server.log_format) {
+            case LOG_FORMAT_LOGFMT:
+                const char * role;
```

Comment:
There are 6 combinations of log_format X role. Could be more moving forward. This is another way to code the logic, but I'm not sure if it's the "better" way...

Reply:
But it does keep the minimum memory usage on role print.